README.md

```diff
@@ -1,57 +1,179 @@
+
+This repo offers guidance on securing Azure Storage account through a private endpoint while still allowing IoT Hub to interact with it securely, as well as allowing client IoT devices to reach the account through a gateway (Application Gateway and optionally Azure Firewall). With this approach the Storage account no longer allows public Internet access, which is a common requirement in enterprise deployments.
 
 ## Features
 
-This project framework provides the following features:
+This project provides different options for configuring Azure IoT Hub File upload to a Storage private endpoint:
 
-* Feature 1
-* Feature 2
-* ...
+* [Documentation](./docs/doc-iothub-storage-private.md) with guidance an details on how you can accomplish this configuration in a Hub-Spoke network topology, leveraging Application Gateway, and Azure Firewall for traffic inspection
+* Quickstart: Azure CLI based step-by-step scripts with a simplified setup (single Virtual Network & Resource Group) and sample client app
 
 ## Getting Started
 
 ### Prerequisites
 
-(ideally very short, if any)
+* Azure subscription
+* Azure CLI
+* Visual Studio Code
+* OpenSSL CLI
+* Bash terminal, on Windows you can use WSL
+* .NET SDK 8 (if you wish to run the client and server samples for file upload)
+* DNS provider if you wish to test HTTPS with your custom domain, and SSL Certificate for your custom domain
 
-- OS
-- Library version
-- ...
+> [!TIP]
+> Without a custom DNS and an SSL certificate available, you can still test the entire setup but the IoT client device portion of this quickstart will not be completed.
 
-### Installation
+### Quickstart
 
-(ideally very short)
+This quickstart provides instructions on how to set up Azure IoT Hub's file upload functionality through an Azure Storage account that only allows private connections, through step-by-step scripts. The scripts deploy and configure a simplified set of Azure resources to showcase Azure Storage with private endpoint, Application Gateway and Azure IoT Hub routing to Azure Storage. The Storage account disables any public Internet access and only accessible within the Virtual Network. In this quickstart there is no Azure Firewall for traffic inspection.
+Finally, two sample .NET apps interact with the resources deployed to showcase the end to end data flows.
 
-- npm install [package name]
-- mvn install
-- ...
+> [!WARNING]
+> This sample currently generates self-signed certificate and key files on disk. This should be improved to leverage Azure Key Vault for securely storing and retrieving the password, certificate and key. Application Gateway supports retrieving an SSL certificate from a Key Vault account.
 
-### Quickstart
-(Add steps to get up and running quickly)
+#### Setup
 
-1. git clone [repository clone url]
-2. cd [repository name]
-3. ...
+* Open a Bash terminal
+* `git clone` this repo
+* cd [repository name]
```

Comment:
Use code fence blocks for commands.  in this case, it would be something like...

~~~suggestion
1. Clone this repo and go to the root directory.

   ```bash
   git clone https://github.com/Azure-Samples/azure-edge-extensions-iothub-fileupload-privatelink
   cd azure-edge-extensions-iothub-fileupload-privatelink
   ```
~~~